src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3955,8 +3929,7 @@ class TransactionStatementSegment(BaseSegment):
     https://docs.snowflake.com/en/sql-reference/sql/rollback.html
     """
 
-    type = "transaction_statement"
-    match_grammar = OneOf(
+    match_grammar = OneOf(  # type: ignore
```

Comment:
Oh so this is because it moved form a `Sequence` to a `OneOf` is that it?

Hmm... that could be quite annoying then for people adding new grammar.

Reply:
I wonder if there's a solution. Mypy likes to guess types. It's annoying here because I think BaseSegment has the correct, general type, but mypy wants to enforce the more specific type it got from the ANSI definition. The type checker is great most of the time, but sometimes annoying.

Reply:
According to this, the behavior is by design.

Note that you're not required to use inheritance, and in many cases, there's little or no advantage to doing so.

It's really useful for those segment types such as object names that add their own functions that are used various places, including the rules.

Any thoughts on this overall space? I started down this path because of a segment type that was missing a function defined on the ANSI "base", causing a runtime error in a rule. Since then, I've been exploring the possibilities on the core side (new/better error checks) and on the dialect files (shorter code, getting context-sensitive code help from editors that support type annotations such as PyCharm).

Happy to take or leave bits of this -- the goal is to simplify and reduce bugs, which can be a balancing act since the underlying language and tools have limits and quirks.